litt/cli/push.go

```diff
@@ -0,0 +1,366 @@
+		}
+	} else if deleteAfterTransfer {
+		return fmt.Errorf("--no-gc is required when pushing a non-snapshot table")
+	}
+
+	// Ensure the remote segment directories exists.
+	for _, dest := range destinations {
+		segmentDir := path.Join(dest, tableName, segment.SegmentDirectory)
+		err = connection.Mkdirs(segmentDir)
+		if err != nil {
+			return fmt.Errorf("failed to create segment directory %s at destination %s: %w",
+				segmentDir, dest, err)
+		}
+	}
+
+	// Used to limit rsync concurrency.
+	rsyncLimiter := make(chan struct{}, threads)
+
+	rsyncsInProgress := atomic.Int64{}
+
+	// Transfer the files.
+	for i := lowestSegmentIndex; i <= highestSegmentIndex; i++ {
+		seg := segments[i]
+		filesToTransfer := seg.GetFilePaths()
+
+		for _, filePath := range filesToTransfer {
+			fileName := path.Base(filePath)
+
+			destination := ""
+			if existingDest, exists := existingFilesMap[fileName]; exists {
+				destination = existingDest
+			} else {
+				destination, err = determineDestination(fileName, destinations)
+				if err != nil {
+					return fmt.Errorf("failed to determine destination for file %s: %w", fileName, err)
+				}
+			}
+
+			targetLocation := path.Join(destination, tableName, segment.SegmentDirectory, fileName)
+
+			rsyncLimiter <- struct{}{}
+			rsyncsInProgress.Add(1)
+
+			boundFilePath := filePath
+			go func() {
+				err = connection.Rsync(boundFilePath, targetLocation, throttleMB)
+				if err != nil {
+					errorMonitor.Panic(err)
+				}
+				<-rsyncLimiter
+				rsyncsInProgress.Add(-1)
+			}()
+		}
+	}
+
+	// Wait for all rsyncs to complete.
+	for rsyncsInProgress.Load() > 0 {
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	// Check if there were any errors during the transfer.
+	if ok, err := errorMonitor.IsOk(); !ok {
+		return fmt.Errorf("error detected during transfer: %w", err)
+	}
+
+	// Now that we have transferred the files, we can delete them if requested.
+	if deleteAfterTransfer {
+
+		err = deleteLocalSegments(segments, tableName, isSnapshot, sources, destinations, highestSegmentIndex)
+		if err != nil {
+			return fmt.Errorf("failed to delete segments after transfer: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// Deletes local segments after they have been successfully transferred to the remote destination(s).
+func deleteLocalSegments(
+	segments map[uint32]*segment.Segment,
+	tableName string,
+	isSnapshot bool,
+	sources []string,
+	destinations []string,
+	highestSegmentIndex uint32) error {
+
+	// Delete the segments.
+	for _, seg := range segments {
+		seg.Release()
+	}
+	// Wait for deletion to complete.
+	for _, seg := range segments {
+		err := seg.BlockUntilFullyDeleted()
+		if err != nil {
+			return fmt.Errorf("failed to delete segment %d for table %s: %w",
+				seg.SegmentIndex(), tableName, err)
+		}
+	}
+
+	if isSnapshot {
+		// If we are dealing with a snapshot, update the lower bound file.
+		boundaryFile, err := disktable.LoadBoundaryFile(true, path.Join(sources[0], tableName))
```

Comment:
You're using a raw bool parameter, when you have a `disktable.BoundaryType` defined. Also elsewhere in this file.